Engine/Shared/DefaultEventHandlers/PlaySounds/PlaySoundsHandler.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type
+** File description:
+** PlaySoundsHandler.cpp
+*/
+
+#include "PlaySoundsHandler.hpp"
+
+PlaySoundsHandler::PlaySoundsHandler(std::string eventType)
+    : AEventHandler("PlaySounds") {}
+
+bool PlaySoundsHandler::processEvent(std::shared_ptr<IEvent> event,
+                                      std::shared_ptr<SceneManager::ISceneManager> sceneManager,
+                                      std::shared_ptr<ABINetwork::INetworkUnit> networkUnit,
+                                      std::shared_ptr<IGraphic> graphicLib)
+{
+    try {
+        int key = graphicLib->getKeyDownInput();
+        if (graphicLib->isMouseButtonDown(IGraphic::MouseButtons::MOUSE_LEFT))
+            key = MOUSE_LEFT_CLICK;
+        if (graphicLib->isMouseButtonDown(IGraphic::MouseButtons::MOUSE_RIGHT))
+            key = MOUSE_RIGHT_CLICK;
+        if (graphicLib->isMouseButtonDown(IGraphic::MouseButtons::MOUSE_MIDDLE))
+            key = MOUSE_MIDDLE_CLICK;
+        if (sceneManager->getSoundMap().contains(static_cast<KEY_MAP>(key)))
+            graphicLib->playSound(sceneManager->getSoundMap().at(static_cast<KEY_MAP>(key)));
```

Comment:
It involves that you are forced to make a left, right or middle click to play a sound, so you can't just play a sound by pressing a key, like a spell for example or anything. You should have made this handler like the music handler, just receiving a path and playing the sound. This handler would have been great but not as a default one